src/bsoncxx/oid.hpp

```diff
@@ -104,7 +107,10 @@ class BSONCXX_API oid {
     ///
     /// Conversion operator that indicates that the oid is initialized.
     ///
-    /// @return True if the oid is initialized.
+    /// @deprecated
```

Comment:
Thanks for doing the doxygen deprecation. Please note that there is also a macro MONGOCXX_DEPRECATED that will expand to __attribute__((__deprecated__))), which will cause deprecation warnings for users. Can we also add that here, and on the deprecated constructor?